skymp5-server/cpp/server_guest_lib/WorldState.cpp

```diff
@@ -293,6 +330,18 @@ bool WorldState::AttachEspmRecord(const espm::CombineBrowser& br,
     return false;
   }
 
+  espm::LookupResult res = br.LookupById(worldOrCell);
+  uint16_t cellFlags =
+    reinterpret_cast<espm::CELL*>(res.rec)->GetData(cache).flags;
+  bool isInterior = cellFlags & espm::CELL::Flags::Interior;
+  bool isExterior = !isInterior;
+  const NpcSettingsEntry& entry = npcSettings[espmFiles[GetFileIdx(baseId)]];
```

Comment:
maybe would be better to not rely on default values here, but use iterators instead. and if not found in npcSettings, set needed values explicitly.

espmFiles may also not contain such index for some weird reason.

let's add some checks and logging